hscontrol/types/node.go

```diff
@@ -196,55 +201,41 @@ func (node *Node) HasIP(i netip.Addr) bool {
 	return false
 }
 
-// IsTagged reports if a device is tagged
-// and therefore should not be treated as a
-// user owned device.
-// Currently, this function only handles tags set
-// via CLI ("forced tags" and preauthkeys).
+// IsTagged reports if a device is tagged and therefore should not be treated
+// as a user-owned device.
+// When a node has tags, the tags define its identity (not the user).
 func (node *Node) IsTagged() bool {
-	if len(node.ForcedTags) > 0 {
+	// Check both node.Tags and AuthKey.Tags for backwards compatibility
+	// During migration, some nodes might have tags on PreAuthKey but not yet on node
+	if len(node.Tags) > 0 {
 		return true
 	}
 
 	if node.AuthKey != nil && len(node.AuthKey.Tags) > 0 {
 		return true
 	}
 
-	if node.Hostinfo == nil {
-		return false
-	}
-
-	// TODO(kradalby): Figure out how tagging should work
-	// and hostinfo.requestedtags.
-	// Do this in other work.
-
 	return false
 }
 
-// HasTag reports if a node has a given tag.
-// Currently, this function only handles tags set
-// via CLI ("forced tags" and preauthkeys).
-func (node *Node) HasTag(tag string) bool {
-	return slices.Contains(node.Tags(), tag)
+// IsUserOwned returns true if node is owned by a user (not tagged).
+// Tagged nodes may have a UserID for "created by" tracking, but the tag is the owner.
+func (node *Node) IsUserOwned() bool {
+	return !node.IsTagged()
 }
 
-func (node *Node) Tags() []string {
-	var tags []string
-
-	if node.AuthKey != nil {
-		tags = append(tags, node.AuthKey.Tags...)
+// HasTag reports if a node has a given tag.
+func (node *Node) HasTag(tag string) bool {
+	// Check both node.Tags and AuthKey.Tags for backwards compatibility
+	if slices.Contains(node.Tags, tag) {
+		return true
 	}
 
-	// TODO(kradalby): Figure out how tagging should work
-	// and hostinfo.requestedtags.
-	// Do this in other work.
-	// #2417
-
-	tags = append(tags, node.ForcedTags...)
-	sort.Strings(tags)
-	tags = slices.Compact(tags)
+	if node.AuthKey != nil && slices.Contains(node.AuthKey.Tags, tag) {
+		return true
+	}
```

Comment:
Same as comment above, used tags should only exist on the node structure, so we need to remove this dependency. The authkey tags are only relevant at the time of authentication.